clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/orchestration/DefaultOrchestrationProcessor.groovy

```diff
@@ -201,4 +212,42 @@ class DefaultOrchestrationProcessor implements OrchestrationProcessor {
     summary.putAll(additionalFields)
     return summary
   }
+
+  @Nonnull
+  private GetTaskResult getTask(String clientRequestId) {
+    def existingTask = taskRepository.getByClientRequestId(clientRequestId)
+    if (existingTask) {
+      if (!isTaskRetryable(existingTask)) {
+        return new GetTaskResult(existingTask, false)
+      }
+      existingTask.updateStatus(TASK_PHASE, "Re-initializing Orchestration Task...")
+      return new GetTaskResult(existingTask, true)
+    }
+    return new GetTaskResult(
+      taskRepository.create(TASK_PHASE, "Initializing Orchestration Task...", clientRequestId),
+      true
+    )
+  }
+
+  private void failTask(@Nonnull Task task, @Nonnull Exception e) {
+    if (task.getSagaIds().isEmpty()) {
+      // Tasks that are not Saga-backed are automatically assumed to be not retryable.
```

Comment:
❤️ this comment

Reply:
also minor but I think this would read a bit nicer without the low-level-ish list check. Maybe some sweet sugar like `task.hasSagas()` or something